src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2549,6 +2560,100 @@ private function registerUidConfiguration(array $config, ContainerBuilder $conta
+                $def->addMethodCall('blockElement', [$element], true);
+            }
+
+            foreach ($sanitizerConfig['drop_elements'] ?? [] as $element) {
+                $def->addMethodCall('dropElement', [$element], true);
+            }
+
+            // Configures attributes
+            foreach ($sanitizerConfig['allow_attributes'] ?? [] as $attribute => $elements) {
+                $def->addMethodCall('allowAttribute', [$attribute, $elements], true);
+            }
+
+            foreach ($sanitizerConfig['drop_attributes'] ?? [] as $attribute => $elements) {
+                $def->addMethodCall('dropAttribute', [$attribute, $elements], true);
+            }
+
+            // Force attributes
+            foreach ($sanitizerConfig['force_attributes'] ?? [] as $element => $attributes) {
+                foreach ($attributes as $attrName => $attrValue) {
+                    $def->addMethodCall('forceAttribute', [$element, $attrName, $attrValue], true);
+                }
+            }
+
+            // Settings
+            if (isset($sanitizerConfig['force_https_urls'])) {
+                $def->addMethodCall('forceHttpsUrls', [$sanitizerConfig['force_https_urls']], true);
+            }
+
+            if (isset($sanitizerConfig['allowed_link_schemes'])) {
+                $def->addMethodCall('allowLinkSchemes', [$sanitizerConfig['allowed_link_schemes']], true);
+            }
+
+            if (isset($sanitizerConfig['allowed_link_hosts'])) {
+                $def->addMethodCall('allowLinkHosts', [$sanitizerConfig['allowed_link_hosts']], true);
+            }
+
+            if (isset($sanitizerConfig['allow_relative_links'])) {
+                $def->addMethodCall('allowRelativeLinks', [$sanitizerConfig['allow_relative_links']], true);
+            }
+
+            if (isset($sanitizerConfig['allowed_media_schemes'])) {
+                $def->addMethodCall('allowMediaSchemes', [$sanitizerConfig['allowed_media_schemes']], true);
+            }
+
+            if (isset($sanitizerConfig['allowed_media_hosts'])) {
+                $def->addMethodCall('allowMediaHosts', [$sanitizerConfig['allowed_media_hosts']], true);
+            }
+
+            if (isset($sanitizerConfig['allow_relative_medias'])) {
+                $def->addMethodCall('allowRelativeMedias', [$sanitizerConfig['allow_relative_medias']], true);
+            }
+
+            // Custom attribute sanitizers
+            foreach ($sanitizerConfig['with_attribute_sanitizers'] ?? [] as $serviceName) {
+                $def->addMethodCall('withAttributeSanitizer', [new Reference($serviceName)], true);
+            }
+
+            foreach ($sanitizerConfig['without_attribute_sanitizers'] ?? [] as $serviceName) {
+                $def->addMethodCall('withoutAttributeSanitizer', [new Reference($serviceName)], true);
+            }
+
+            // Create the sanitizer and link its config
+            $container->register($sanitizerName, HtmlSanitizer::class)
```

Comment:
Not adding a prefix for the service seems weird to me. When we have named configuration in Symfony and in bundles, we generally don't require uniqueness over the whole container. So this makes it easy to override another service by mistake.

I suggest registering them as `html_sanitizer.sanitizer.<name>` instead, with `html_sanitizer.config.<name>` for the config.

Reply:
we do so on httpclient and it works well
we also do add a suffix for workflows, and it forces a needless convention on named autowiring aliases
I prefer not forcing any convention and showing examples to ppl, eg `my.sanitizer` as name

Reply:
I'm not asking to change the autowiring aliases here. Only the service names.

And if we require those names to avoid conflicts with any service in the container, we must document it explicitly in the info of the configuration, which is not the case right now.

Reply:
I reopened https://github.com/symfony/symfony/issues/41198#issuecomment-1002542381 so that we can have the discussion

Reply:
I fixed this by renaming services to:

```
html_sanitizer.sanitizer.my.sanitizer
html_sanitizer.config.my.sanitizer
```

And keeping the named autowiring alias to `$mySanitizer`